Teapot/MockTeapot.swift

```diff
@@ -22,38 +20,56 @@ open class MockTeapot: Teapot {
         case invalidMockFile(String)
     }
 
-    /// The status code the URL request should return
+    private let currentBundle: Bundle
+    private let resource: String
     private let statusCode: StatusCode
 
+    private var overrideEndpointDictionary = [String: String]()
+
     /// Initialiser.
     ///
     /// - Parameters:
-    ///   - bundle: the bundle of your test target. When you add a json file with the name of the endpoint to your test target it will return this data.
+    ///   - bundle: the bundle of your test target, where it will search for the mock file
+    ///   - mockFileName: the name of the mock file containing the json that will be returned
     ///   - statusCode: the status code for the response to return errors. Default is 200 "ok" 👌
-    public init(bundle: Bundle, statusCode: StatusCode = .ok) {
+    public init(bundle: Bundle, mockFileName: String, statusCode: StatusCode = .ok) {
         self.currentBundle = bundle
+        self.resource = mockFileName
         self.statusCode = statusCode
 
         super.init(baseURL: URL(string: "https://mock.base.url.com")!)
     }
 
+    /// overrideEndPoint.
+    /// call this method if you need a call to a certain endpoint to return a specified mock file
+    /// for example when you have a security call to the server that get's called every time you do an APICall
+    ///
+    /// - Parameters:
+    ///   - endPoint: the endpoint tht needs to get overridden
+    ///   - fileName: the name of the json file from which you want the data to be returned
+    public func overrideEndPoint(_ endPoint: String, withFileName fileName: String) {
```

Comment:
Even after reading the doc above, I'm not clear yet on what this method is doing.

Reply:
You think this is more clear? `set the filename of the mocked json you want to return for a call to a certain endpoint`

It's a bit vague I guess. In toshi for example we do a call to the api to get the timestamp before the other call gets performed. So i need to override this call to get the timestamp to return a certain mocked json file.

Reply:
Yeah still vague. Sounds like a good case for more descriptive longer docs, with maybe some examples. I haven't dealt with these in a long time, not sure how the formatting is done anymore.